src/bp-messages/classes/class-bp-messages-thread.php

```diff
@@ -942,7 +1021,121 @@ public static function get_total_threads_for_user( $user_id, $box = 'inbox', $ty
 
 		$bp = buddypress();
 
-		return (int) $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(thread_id) FROM {$bp->messages->table_name_recipients} WHERE user_id = %d AND is_deleted = 0 {$exclude_sender} {$type_sql}", $user_id ) );
+		return (int) $wpdb->get_var(
+			$wpdb->prepare( "SELECT COUNT(thread_id) FROM {$bp->messages->table_name_recipients} WHERE user_id = %d AND is_deleted = 0 {$exclude_sender} {$type_sql}", $user_id )
+		);
+	}
+
+	/**
+	 * Returns the total number of messages in a thread.
+	 *
+	 * @since 11.0.0
+	 *
+	 * @global BuddyPress $bp The one true BuddyPress instance.
+	 * @global wpdb $wpdb WordPress database object.
+	 *
+	 * @param int $thread_id The message thread ID.
+	 * @return int Total thread message count
+	 */
+	public static function get_total_thread_message_count( $thread_id ) {
+		global $wpdb;
+
+		$total_thread_message_count = wp_cache_get( $thread_id, 'bp_messages_unread_count' );
+
+		if ( false === $total_thread_message_count ) {
+			$bp = buddypress();
+
+			$total_thread_message_count = (int) $wpdb->get_var(
+				$wpdb->prepare( "SELECT COUNT(id) FROM {$bp->messages->table_name_messages} WHERE thread_id = %d", $thread_id )
+			);
+
+			wp_cache_set( $thread_id, $total_thread_message_count, 'bp_messages_thread_count' );
+		}
+
+		/**
+		 * Thread messages count.
+		 *
+		 * @since 11.0.0
+		 *
+		 * @param int $total_thread_message_count Total thread messages count.
+		 * @param int $thread_id                  ID of the thread.
+		 */
+		return apply_filters( 'messages_thread_get_total_message_count', (int) $total_thread_message_count, (int) $thread_id );
+	}
+
+	/**
+	 * Returns the total number of recipients in a thread.
+	 *
+	 * @since 11.0.0
+	 *
+	 * @global BuddyPress $bp The one true BuddyPress instance.
+	 * @global wpdb $wpdb WordPress database object.
+	 *
+	 * @param int $thread_id The message thread ID.
+	 * @return int Total thread recipients count.
+	 */
+	public static function get_total_thread_recipients_count( $thread_id ) {
+		global $wpdb;
+
+		$total_recipients_count = wp_cache_get( $thread_id, 'bp_messages_recipients_count' );
```

Comment:
This new cache group should probably be setup into this function:
https://github.com/buddypress/buddypress/blob/master/src/bp-messages/classes/class-bp-messages-component.php#L417|L428

I believe this was the initial purpose of: https://buddypress.trac.wordpress.org/ticket/8590

Reply:
Good cache!